nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs

```diff
@@ -962,13 +962,20 @@ private static void ThrowOnMissingPackages(string output)
         }
     }
 
-    private static void ThrowOnUnresolvableDependencies(string output)
+    private static void ThrowOnUpdateNotPossible(string output)
     {
-        var unresolvablePackagePattern = new Regex(@"Unable to resolve dependencies\. '(?<PackageName>[^ ]+) (?<PackageVersion>[^']+)'");
-        var match = unresolvablePackagePattern.Match(output);
-        if (match.Success)
+        var patterns = new[]
         {
-            throw new UpdateNotPossibleException([$"{match.Groups["PackageName"].Value}.{match.Groups["PackageVersion"].Value}"]);
+            new Regex(@"Unable to resolve dependencies\. '(?<PackageName>[^ ]+) (?<PackageVersion>[^']+)'"),
```

Comment:
Please pass a timeout to all Regex's. Also consider making them static and compiled (although given that this is an error case and not a hot path that probably doesn't matter too much).